Add code tabs to type checker page (#2726)
scala/docs.scala-lang: _overviews/scala3-migration/incompat-type-checker.md

The Scala 3 compiler does not permit this anymore.

{% tabs scala-2-unsound_vc_1 %}
{% tab 'Scala 2 Only' %}
~~~ scala
class Foo[-A](x: List[A]) {
  def f[B](y: List[B] = x): Unit = ???
}

class Outer[+A](x: A) {
  class Inner(y: A)
}
~~~
{% endtab %}
{% endtabs %}

> So if you compile in Scala 3, you should get this kind of error.

Reviewer: ```suggestion
So if you compile in Scala 3, you will get the following error.
```

The Scala 3 compiler does not permit this anymore.

{% tabs scala-3-unsound_vc_1 %}

> {% tab 'Scala 3 Only' %}

Reviewer: There is confusion going on here, this code is not valid in Scala 3, so we should put “Scala 2 Only” in the tab title.

~~~ scala
class Foo[-A](x: List[A]) {
  def f[B](y: List[B] = x): Unit = ???
}

class Outer[+A](x: A) {
  class Inner(y: A)
}
~~~

> ~~~ text

Reviewer: Did you check that the rendering is correct? IIRC, we had to use the `highlight` Jekyll directive for some reason.

In this example, we can relax the constraint on `x` and `y` by stating that `A` is a common ancestor of both type arguments.
This makes the compiler type-check the code successfully.
{% tabs scala-3-unsound_pm_2 %}
{% tab 'Scala 3 Only' %}
~~~ scala

> def combineFetch[A](x: Fetch[_ <: A], y: Fetch[_ <: A]): Fetch[A] = Fetch(x.ids ++ y.ids)

Reviewer: Is this snippet not “Scala 2 and 3”?

~~~

> Alternatively, a general but unsafe solution is to cast.

Reviewer: This sentence should stay outside the tab.
